spec/controllers/article_controller_spec.rb

```diff
@@ -0,0 +1,209 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe ArticlesController, type: :controller do
+  describe 'GET #index' do
+    subject { get :index }
+    include_examples 'not_authorized'  
+
+    context 'when login' do
+      let(:articles) { create_list(:article, 4) }
+      let(:user) { create(:user) }
+      before { sign_in user }
+        
+      it 'return 200' do
+        expect(subject.status).to eq(200)
+      end
+
+      it 'return data' do
+        subject
+        expect(assigns(:articles)).to match_array(articles)
+      end      
+    end
+  end
+
+  describe 'GET #show' do
+    subject { get :show, params: params }
+    include_examples 'not_authorized'
+
+    let(:article) { create(:article) }
+    let(:params) { {id: article.id} }
+
+    context 'when login' do
+      let(:user) { create(:user) }
+      before { sign_in user }
+
+      it 'return 200' do
+        expect(subject.status).to eq(200)
+      end
+
+      it 'return data' do
+        subject
+        expect(assigns(:article)).to eq(article)
+      end
+
+      it 'not found' do
+        params.update({id: 'invalid'})
```

Comment:
This way might be a little bit faster, it might not seem to be necessary for now, but speaking of a whole test suites, it's going to be significantly faster (not like in cookiewow 😢) 
```suggestion
        params['id'] = 'invalid'
```